js/main.js

```diff
@@ -219,45 +226,67 @@ function setupThenOpenMainWindow() {
     processProtocolAction(process.argv);
 
     isAppAlreadyOpen = true;
-
-    // allows installer to launch app and set appropriate global / user config params.
-    let hasInstallFlag = getCmdLineArg(process.argv, '--install', true);
-    let perUserInstall = getCmdLineArg(process.argv, '--peruser', true);
-    let customDataArg = getCmdLineArg(process.argv, '--userDataPath=', false);
-
-    if (customDataArg && customDataArg.split('=').length > 1) {
-        let customDataFolder = customDataArg.split('=')[1];
-        app.setPath('userData', customDataFolder);
-    }
-    if (!isMac && hasInstallFlag) {
-        getConfigField('launchOnStartup')
-            .then(setStartup)
-            .then(() => updateUserConfigWin(perUserInstall))
-            .then(app.quit)
-            .catch(app.quit);
-        return;
-    }
-
-    // allows mac installer to overwrite user config
-    if (isMac && hasInstallFlag) {
-        // This value is being sent from post install script
-        // as the app is launched as a root user we don't get
-        // access to the config file
-        let launchOnStartup = process.argv[3];
-        // We wire this in via the post install script
-        // to get the config file path where the app is installed
-        setStartup(launchOnStartup)
-            .then(updateUserConfigMac)
-            .then(app.quit)
-            .catch(app.quit);
-        return;
-    }
-
     getUrlAndCreateMainWindow();
-
+    
     // Event that fixes the remote desktop issue in Windows
     // by repositioning the browser window
     electron.screen.on('display-removed', windowMgr.verifyDisplays);
+    
+}
+
+function checkFirstTimeLaunch() {
+    
+    return new Promise((resolve) => {
+
+        getUserConfigField('version')
+            .then((configVersion) => {
+
+                const appVersionString = app.getVersion().toString();
+
+                const execPath = nodePath.dirname(app.getPath('exe'));
+                const shouldUpdateUserConfig = execPath.indexOf('AppData/Local/Programs') !== -1 || isMac;
+
+                if (configVersion
+                    && typeof configVersion === 'string'
+                    && (compareSemVersions.check(appVersionString, configVersion) === 0
+                    || compareSemVersions.check(appVersionString, configVersion) === -1)) {
+                    return resolve();
+                }
+
+                if (shouldUpdateUserConfig) {
```

Comment:
can we combine the comparison of version check logic with the exe path and have one single if condition?